content/tutorials/max-gas-pub-data/10.index.md

```diff
@@ -0,0 +1,421 @@
+  - Eliminates caps on `gasPerPubdata.`
+
+## Implications for Users and Developers
+
+1. **Cost Predictability**: While costs may vary with L1 gas prices, users can estimate costs based on the state changes their transactions will cause.
+2. **Optimization Opportunities**: Developers can optimize their applications to minimize state changes, potentially reducing users' costs.
+3. **Efficient for Certain Use Cases**: Applications like oracles or high-frequency trading platforms may find this model particularly cost-effective.
+4. **Transaction Behavior**: Users should be aware that transactions may be rejected or reverted based on
+pubdata costs, even if they seem to have sufficient gas for execution.
+5. **Flexible Pricing**: The absence of hard caps on `gasPerPubdata` allows for more flexible pricing models.
+
+The following section provides practical insights into measuring gas costs and setting the DEFAULT_GAS_PER_PUBDATA_LIMIT in the ZKSync era.
+
+## Measuring Gas Costs and Setting Different Values for Pubdata Gas Limit
+
+This guide demonstrates how to measure gas costs and set the `DEFAULT_GAS_PER_PUBDATA_LIMIT` in the ZKSync Era using a practical
+example: a ZKFest Voting Contract. **What is Max Gas Per Pubdata?** Max gas per pubdata is a value attached to each transaction
+on ZKsync Era, representing the maximum amount of gas a user is willing to pay for each byte of pubdata (public data) published on Ethereum.
+
+**Key points:**
+
+1. Default value: 50,000 gas per pubdata byte
+2. Can be customized per transaction; in this case, we're testing three cases
+3. Affects transaction success and cost
+
+## Objective
+
+We'll deploy and interact with a smart contract using various gas settings to understand how different parameters affect
+transaction execution and costs on ZKSync Era.
+
+## Running the Experiment
+
+1. Set up your environment by creating a new project with ZKsync CLI
+2. Create the ZKFestVoting contract and the deploy script based on the code below
+3. Analyze the output for each scenario, paying attention to:
+    - Successful transactions and their gas usage
+    - Rejected transactions and their error messages
+
+## Step 1: Smart Contract (ZKFestVoting.sol)
+
+```solidity [ZKFestVoting.sol]
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+contract ZKFestVoting {
+    enum Stage { Culture, DeFi, ElasticChain }
+    mapping(address => uint8) public participation;
+    event Voted(address indexed voter, Stage stage);
+
+    function vote(string memory stageName) external {
+        uint256 stageIndex = getStageIndex(stageName);
+        require(stageIndex < 3, "Invalid stage");
+        uint256 stageBit = 1 << stageIndex;
+        require((participation[msg.sender] & stageBit) == 0, "Already voted for this stage");
+        participation[msg.sender] |= uint8(stageBit);
+        emit Voted(msg.sender, Stage(stageIndex));
+    }
+
+    // Helper functions omitted for brevity
+}
+```
+
+::drop-panel
```

Comment:
FYI: this one renders badly in the markdown github preview, but probably it is not supposed to